experimental/token/raw.go

```diff
@@ -99,18 +104,41 @@ type nat struct {
 	// than the start, it makes adding tokens one by one to the stream
 	// easier, because once the token is pushed, its start and end are
 	// set correctly, and don't depend on the next token being pushed.
-	end           uint32
-	kindAndOffset int32
+	end uint32
+
+	// This contains compressed metadata about a token in the following format:
+	//
+	// 1. Bits [0:3] is the Kind.
+	// 2. Bit [3] is whether this is a non-leaf.
+	//    a. If it is a non-leaf, bits [4:6] determine the Keyword value if
```

Comment:
```suggestion
	// 2. Bit [4] is whether this is a non-leaf.
	//    a. If it is a non-leaf, bits [5:6] determine the Keyword value if
```

Reply:
No this isn't right, the range here is exclusive. Added a comment explaining that.